src/telemetry/TelemetryHelper.cpp

```diff
@@ -0,0 +1,62 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file TelemetryHelper.h
+ @brief Define a helper class, used for telemetry initialization.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "TelemetryHelper.h"
+
+#include "TempDirectory.h"
+
+#include "lib-telemetry/TelemetryManager.h"
+#include "lib-string-utils/CodeConversions.h"
+
+#include "TelemetryDialog.h"
+
+static const char* prefsKeyEnabled = "/Telemetry/Enabled";
+static const char* prefsKeyDialogShown = "/Telemetry/PersmissionDialogShown";
+
+TelemetryHelper::TelemetryHelper ()
+{
+    using namespace audacity;
+
+    if (!gPrefs->ReadBool (prefsKeyDialogShown, false))
+    {
+        TelemetryDialog dlg (nullptr);
+        const int code = dlg.ShowModal ();
+
+        gPrefs->Write (prefsKeyDialogShown, true);
+        gPrefs->Write (prefsKeyEnabled, code == wxID_YES);
+    }
+
+    telemetry::Initialize (
+        "Audacity",
+        ToUTF8 (AUDACITY_VERSION_STRING),
+        ToUTF8 (TempDirectory::TempDir ()),
+        gPrefs->ReadBool(prefsKeyEnabled, false)
+    );
+
+#ifdef GOOGLE_UA_TRACKING_ID
+    if (*(GOOGLE_UA_TRACKING_ID) != '\0')
+        telemetry::SetTelemetryService (telemetry::TelemetryService::GoogleAnalytics_UA, GOOGLE_UA_TRACKING_ID);
+#endif
+
+#ifdef METRICA_TRACKING_ID
+    if (*(METRICA_TRACKING_ID) != '\0')
+        telemetry::SetUserTrackingService (telemetry::UserTrackingService::YandexMetrica, METRICA_TRACKING_ID);
+#endif
+}
+
+TelemetryHelper::~TelemetryHelper ()
+{
+    audacity::telemetry::Terminate ();
+}
+
+void TelemetryHelper::UpdatePrefs ()
+{
+    audacity::telemetry::SetTelemetryEnabled (gPrefs->ReadBool (prefsKeyEnabled, true));
```

Comment:
This line seems to default the telemetry to be enabled when the config key doesn't exist. Not having a config key is not a consent to be tracked.

Reply:
FWIW I'm currently discussing this in the Audacity discord with the OP. This appears to be a workaround for a UI bug, to be completely fair. 

https://github.com/audacity/audacity/blob/2c14a12b4f3ded3b24181f2b08f32c71ddbaea01/src/telemetry/TelemetryHelper.cpp#L31

That is where the telemetry is being initialized - you can see it default to `false`.

Why it's `true` here is unclear and I'm pushing for a proper comment explanation (or even better, a fix) to be added.

Good catch here though.